tailnet/controllers.go

```diff
@@ -845,30 +854,48 @@ type workspace struct {
 	agents map[uuid.UUID]agent
 }
 
+// addAllDNSNames adds names for all of its agents to the given map of names
+func (w workspace) addAllDNSNames(names map[dnsname.FQDN][]netip.Addr) error {
+	for _, a := range w.agents {
+		// TODO: technically, DNS labels cannot start with numbers, but the rules are often not
+		//       strictly enforced.
+		// TODO: support <agent>.<workspace>.<username>.coder
+		fqdn, err := dnsname.ToFQDN(fmt.Sprintf("%s.%s.me.coder.", a.name, w.name))
+		if err != nil {
+			return err
+		}
+		names[fqdn] = []netip.Addr{CoderServicePrefix.AddrFromUUID(a.id)}
+	}
+	// TODO: Possibly support <workspace>.coder. alias if there is only one agent
```

Comment:
I think letting you omit the agent is sensible but allowing you to omit the `me`/`username` sounds a bit dangerous. Someone shares their workspace named `dev` with you and the hostname you're using just breaks randomly.

Reply:
Actually i guess you could just only allow omitting `me`, not `username`

Reply:
omitting the username implies `me`.  So if someone shares their workspace `dev` with you, you could never access it with `dev.coder`, regardless of whether you, yourself have a workspace named `dev`.

Reply:
Yeah, nice. I like these aliases, I think there's a lot of value in having short hostnames.